packages/dd-trace/src/opentracing/tracer.js

```diff
@@ -52,8 +52,16 @@ class DatadogTracer {
       ? getContext(options.childOf)
       : getParent(options.references)
 
+    // as per spec, allow the setting of service name through tags
     const tags = {
-      'service.name': this._service
+      'service.name': options?.tags?.service ? String(options.tags.service) : this._service
+    }
+
+    // edge case handling, if a user manually sets a service name different from the tracer's service name
+    // then we need to ensure that the version is set to undefined
+    // in every other scenario version should be set to the tracer's version
```

Comment:
I assume by "manually" you mean when creating a custom span and passing a service name != global service name?
Does this apply to integration service names that differ from global service name (since it's not "manual" but does have the "different service name" effect)

Reply:
@khanayan123 the comment says that we need to do something but why do we need to do it?

Reply:
Updated the comment, the reason we need to do it is to align with unified service tagging spec https://docs.datadoghq.com/getting_started/tagging/unified_service_tagging/?tab=kubernetes

`DD_VERSION (version)
Environment Variable Name: DD_VERSION
Behavior: Adds version tag to all spans that have the same service name as the default service name (aka DD_SERVICE). Spans with a different service name do not get this tag.`

`Configure spans with DD_VERSION to add version to all spans that fall under the service that belongs to the tracer (generally DD_SERVICE). This means that if your service creates spans with the name of an external service, those spans do not receive version as a tag.`

`Adds version tag to all spans that have the same service name as the default service name (aka DD_SERVICE). Spans with a different service name do not get this tag.`